corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js

```diff
@@ -193,6 +193,197 @@ var filterViewModel = function(filterText, saveButton){
     };
 };
 
+var caseListLookupViewModel = function($el, state, saveButton){
+    'use strict';
+    var self = this;
+
+    var ObservableKeyValue = function(obs){
+        this.key = ko.observable(obs.key);
+        this.value = ko.observable(obs.value);
+    };
+
+    var _fireChange = function(){
+        saveButton.fire('change');
+    };
+
+    self.initSaveButtonListeners = function($el){
+        $el.find('input[type=text], textarea').bind('textchange', _fireChange);
+        $el.find('input[type=checkbox]').bind('change', _fireChange);
+        $el.find(".case-list-lookup-icon button").bind("click", _fireChange); // Trigger save button when icon upload buttons are clicked
+    };
+
+    var _remove_empty = function(type){
+        self[type].remove(function(t){
+            var is_blank = (!t.key() && !t.value());
+            return is_blank;
+        });
+    };
+
+    self.add_item = function(type){
+        _remove_empty(type);
+        var data = (type === 'extras') ? {key: '', value: ''} : {key: ''};
+        self[type].push(new ObservableKeyValue(data));
+    };
+
+    self.remove_item = function(type, item){
+        self[type].remove(item);
+        if (self[type]().length === 0){
+            self.add_item(type);
+        }
+        _fireChange();
+    };
+
+    var _trimmed_extras = function(){
+        return _.compact(_.map(self.extras(), function(extra){
+            if (!(extra.key() === "" && extra.value() ==="")){
+                return {key: extra.key(), value: extra.value()};
+            }
+        }));
+    };
+
+    var _trimmed_responses = function(){
+        return _.compact(_.map(self.responses(), function(response){
+            if (response.key() !== ""){
+                return {key: response.key()};
+            }
+        }));
+    };
+
+    self.serialize = function(){
+        var image_path = $el.find(".case-list-lookup-icon input[type=hidden]").val() || null;
+
+        var data = {
```

Comment:
this may be useful: http://knockoutjs.com/documentation/plugins-mapping.html. `ko.mapping.toJS` takes a KO obj and turns it to JS. might be hard to use here since it's tied to the `self` object which includes more attributes